toml/src/main/kotlin/org/rust/toml/completion/CargoTomlDependencyCompletionProvider.kt

```diff
@@ -30,3 +29,39 @@ class CargoTomlDependencyCompletionProvider : CompletionProvider<CompletionParam
         delegate.addCompletionVariants(parameters, context, result)
     }
 }
+
+class CargoTomlSpecificDependencyHeaderCompletionProvider : CompletionProvider<CompletionParameters>() {
+    private val localCompletionProvider = LocalCargoTomlSpecificDependencyHeaderCompletionProvider()
+    private val cratesIoCompletionProvider = CratesIoCargoTomlSpecificDependencyHeaderCompletionProvider()
+
+    private val delegate: CompletionProvider<CompletionParameters>
```

Comment:
Consider introducing a base class for these three providers to avoid code duplication